### PR TITLE
feat: Add d20 roll mechanic with risk/reward system

### PR DESCRIPTION
**Pull request by wraithfive**
## Summary

Implements a new `/d20` command that adds an optional risk/reward mechanic to the gacha system. Players can roll a d20 within 60 minutes of using `/roll` for a chance at powerful buffs or harsh penalties.

### Key Features

- **Visual d20 animation**: 6-frame animated GIF with progressive number reveal
- **Nat 20 (5% chance)**: Grants "Lucky Streak" buff - next roll guaranteed Epic or Legendary
- **Nat 1 (5% chance)**: "Critical Failure" - cooldown extended to 48 hours instead of 24
- **Normal rolls (2-19, 90%)**: No effect
- **60-minute window**: Must use `/d20` within 60 minutes of using `/roll`
- **One-time use**: Can only use `/d20` once per roll cycle
- **Auto-enabled**: Feature automatically available when server has 3+ Epic/Legendary roles

### Implementation Details

**Database**:
- Added `d20_used` and `guaranteed_epic_plus` columns to `user_cooldowns` table
- Liquibase migrations handle schema changes with proper rollback support

**Code Quality**:
- Extracted magic numbers as named constants (D20_ANIMATION_FRAMES, D20_FRAME_DELAY_MS, D20_WINDOW_MINUTES)
- Comprehensive error handling with fallback for animation failures
- Randomized intermediate frames avoid consecutive duplicates for better visual experience
- Production-appropriate logging (DEBUG for animation, INFO for user actions)

**Testing**:
- 14 unit tests covering all scenarios (availability, window validation, nat 1/20/normal outcomes)
- Integration test verifying full buff application flow
- Test coverage for edge cases and error conditions
- All tests passing

**Security**:
- Removed H2 AUTO_SERVER mode (potential security risk)
- Thread-safe recursive animation pattern using JDA's executor pools

### User Experience

Players see their roll through an animated sequence:
1. d20 GIF appears with first random number
2. Five more random numbers flash (600ms each)
3. Final result revealed with appropriate effect message

The `/mycolor` command shows active buffs, d20 window availability, and extended cooldowns.

## Test Plan

- [x] Unit tests for all d20 scenarios (nat 1, nat 20, normal rolls)
- [x] Integration test for buff application
- [x] Window expiration validation
- [x] One-time use enforcement
- [x] Availability checks (requires 3+ Epic/Legendary roles)
- [x] Error handling and fallback behavior
- [x] Database migrations execute successfully
- [x] Manual testing in Discord (see logs showing 8+ successful d20 rolls)

## Breaking Changes

None - feature is purely additive and automatically enables when role requirements are met.

## Files Changed

- `SlashCommandHandler.java`: Core d20 logic and animation (+382 lines)
- `UserCooldown.java`: Added d20 state tracking fields (+30 lines)
- `D20MechanicTest.java`: Comprehensive test suite (+640 lines)
- `010-add-d20-mechanic-columns.xml`: Database migration (+51 lines)
- `CLAUDE.md`: Documentation (+17 lines)
- `application.properties`: Security fix (H2 AUTO_SERVER removal)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/main/java/com/discordbot/SlashCommandHandler.java [hunk 1]
```diff
@@ -563,6 +645,268 @@ private String getRarityEmoji(Rarity rarity) {
...
+    /**
+     * Show animated d20 roll with GIF and progressive text reveal
+     */
+    private void showD20Animation(SlashCommandInteractionEvent event, int finalRoll, String resultType) {
+        // Get GIF URL - use GitHub raw URL for development/testing, production URL from env var
+        String gifUrl;
+        String baseUrl = System.getenv("ADMIN_PANEL_URL");
+        if (baseUrl == null || baseUrl.contains("localhost")) {
+            // Development: use GitHub raw URL so Discord can fetch it
+            gifUrl = "https://raw.githubusercontent.com/wraithfive/playbot/master/src/main/resources/static/images/d20-roll.gif";
```
[the remark this answers is not in the document]
Comment (wraithfive) [reply src/main/java/com/discordbot/SlashCommandHandler.java]: this was done on purpose for dev testing on local machine because discord can't load the gif from local instance
